src/Kiota.Builder/Writers/Go/CodeMethodWriter.cs

```diff
@@ -166,7 +166,17 @@ private void WriteRequestBuilderBody(CodeClass parentClass, CodeMethod codeEleme
             };
             if(!writePrototypeOnly)
                 WriteMethodDocumentation(code, methodName, writer);
-            var parameters = string.Join(", ", code.Parameters.OrderBy(x => x, parameterOrderComparer).Select(p => conventions.GetParameterSignature(p, parentBlock)).ToList());
+
+            var codeParameters = new List<string>();
+            
+            if(code.Kind == CodeMethodKind.RequestExecutor) // default parameter for executors
+                codeParameters.Add($"{ContextVarName} {ContextVarTypeName}");
+            
+            codeParameters.AddRange(code.Parameters.OrderBy(x => x, parameterOrderComparer)
+                .Select(p => conventions.GetParameterSignature(p, parentBlock)).ToList());
+
+            var parameters = string.Join(", ", codeParameters);
+            
```

Comment:
we should be instead:
- changing the name/type of the cancellation token parameter in a refiner
- adding an entry to the comparer so the context goes first
- reverting this change